suite-native/discovery/src/discoveryMiddleware.ts

```diff
@@ -30,11 +32,10 @@ export const prepareDiscoveryMiddleware = createMiddlewareWithExtraDeps(
         );
 
         // If user enables testnets discovery, run it.
-        if (toggleAreTestnetsEnabled.match(action) && !areTestnetsEnabled && device?.state) {
+        if (toggleAreTestnetsEnabled.match(action) && !areTestnetsEnabled) {
             if (isDeviceFirmwareVersionSupported) {
                 dispatch(
                     startDescriptorPreloadedDiscoveryThunk({
-                        deviceState: device.state,
```

Comment:
this has to stay here, otherwise deviceState is not available from the state in startDescriptorPreloadedDiscoveryThunk